test/suite/test_cursor_bound_fuzz.py

```diff
@@ -108,129 +118,262 @@ def dump_key_range(self):
+                    continue
+                op = random.choice(list(operations))
+                if (op is operations.TRUNCATE):
+                    pass
+                elif (op is operations.UPSERT):
+                    self.apply_update(cursor, i, prepare)
+                elif (op is operations.REMOVE):
+                    self.apply_remove(cursor, i, prepare)
+                else:
+                    raise Exception("Unhandled operation generated")
+        self.applied_ops = True
+
+    # As prepare throws a prepare conflict exception we wrap the call to anything that could
+    # encounter a prepare conflict in a try except, we then return the error code to the caller.
+    def prepare_call(self, func):
+        try:
+            ret = func()
+        except wiredtiger.WiredTigerError as e:
+            if wiredtiger.wiredtiger_strerror(wiredtiger.WT_PREPARE_CONFLICT) in str(e):
+                ret = wiredtiger.WT_PREPARE_CONFLICT
             else:
-                raise Exception("Unhandled operation generated")
+                raise e
+        return ret
+
+    # Given a bound this key returns what would be the start of the bounded range for that bound.
+    # e.g. if we have a lower bound then the key would be the lower bound, however if the lower
+    # bound isn't enabled then the lowest possible key would be min_key.
+    # max_key isn't inclusive so we subtract 1 off it.
+    def get_expected_key(self, bound_set, type):
+        if (type == bound_type.LOWER):
+            if (bound_set.lower.enabled):
+                if (bound_set.lower.inclusive):
+                    return bound_set.lower.key
+                return bound_set.lower.key + 1
+            return self.min_key
+        if (bound_set.upper.enabled):
+            if (bound_set.upper.inclusive):
+                return bound_set.upper.key
+            return bound_set.upper.key - 1
+        return self.max_key - 1
+
+    # When a prepared cursor walks next or prev it can skip deleted records internally before
+    # returning a prepare conflict, we don't know which key it got to so we need to validate that
+    # we see a series of deleted keys followed by a prepared key.
+    def validate_deleted_range(self, start_key, end_key, next):
+        if (next):
+            step = 1
+        else:
+            step = -1
+        self.verbose(2, "Walking deleted range from: " + str(start_key) + " to: " + str(end_key))
+        for i in range(start_key, end_key, step):
+            self.verbose(3, "Validating state of key: " + self.key_range[i].to_string())
+            if (self.key_range[i].is_prepared()):
+                return
+            elif (self.key_range[i].is_deleted()):
+                continue
+            else:
+                self.assertTrue(False)
+
+    # Validate a prepare conflict in the cursor->next scenario.
+    def validate_prepare_conflict_next(self, current_key, bound_set):
```

Comment:
we can technically merge the functions `validate_prepare_conflict_prev` and `validate_prepare_conflict_next`

Reply:
Won't be doing this, as we would like to seperate them for easier readability.